src/org/traccar/protocol/WondexFrameDecoder.java

```diff
@@ -40,6 +42,9 @@ protected Object decode(
             if (channel != null) {
                 channel.write(frame);
             }
+            // Pass deviceId to protocol decoder
+            long deviceId = ((Long.reverseBytes((frame.getLong(0)))) >> 32) & 0xFFFFFFFFL;
+            return ChannelBuffers.copiedBuffer("$ID:" + String.valueOf(deviceId), StandardCharsets.US_ASCII);
```

Comment:
This is not a correct way of handling this situation. If you need this frame, you need to pass it to decode as is.